doc/mgmt/SONiC_Management_Framework_sFlow.md

```diff
@@ -0,0 +1,366 @@
+This feature will allow the user to make/show sFlow configuration changes to CONFIG DB
+### 3.2.2 APP DB
+### 3.2.3 STATE DB
+### 3.2.4 ASIC DB
+### 3.2.5 COUNTER DB
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+### 3.3.2 Other Process
+
+
+## 3.4 SyncD
+
+
+## 3.5 SAI
+
+
+## 3.6 User Interface
+### 3.6.1 Data Models
+Supported SONiC YANG URIs available from Swagger WebUI:
+```
+/sonic-sflow:sonic-sflow/SFLOW/GLOBAL
+{
+  "sonic-sflow:GLOBAL": {
+    "admin_state": "up",
+    "polling_interval": 0,
+    "agent_id": "string"
+  }
+}
+
+/sonic-sflow:sonic-sflow/SFLOW_COLLECTOR
+/sonic-sflow:sonic-sflow/SFLOW_COLLECTOR={collector_name}
+{
+  "sonic-sflow:SFLOW_COLLECTOR": [
+    {
+      "collector_name": "string",
+      "collector_ip": "string",
+      "collector_port": 0
+    }
+  ]
+}
+
+/sonic-sflow:sonic-sflow/SFLOW_SESSION
+/sonic-sflow:sonic-sflow/SFLOW_SESSION={ifname}
+{
+  "sonic-sflow:SFLOW_SESSION": [
+    {
+      "admin_state": "up",
+      "ifname": "Ethernet0",
+      "sample_rate": 4400
+    }
+  ]
+}
+```
+
+### 3.6.2 CLI
+sFlow configuration and show commands will be the same as the one available on the host provided by SONiC
+#### 3.6.2.1 Configuration Commands
+All commands are executed in `configuration-view`:
+```
+sonic# configure terminal
+sonic(config)#
+```
+
+##### Enable sFlow
+```
+sonic(config)# sflow enabled
+sonic(config)#
+```
+
+##### Disable sFlow
+```
+sonic(config)# no sflow enabled
+sonic(config)#
+```
+
+##### Add sFlow Collector
+Syntax:
+
+port# : [0 - 65535]
+
+`sflow collector <collector name> <IPv4/IPv6 address> [port #]`
+
+```
+sonic(config)# sflow collector col1 1.1.1.1
+sonic(config)#
+```
+
+##### Add sFlow Collector with port number
+```
+sonic(config)# sflow collector col2 1.1.1.2 port 4451
+sonic(config)#
+```
+
+##### Remove a sFlow Collector
+Syntax:
+
+`no sflow collector <collector name>`
+
+```
+sonic(config)# no sflow collector col1
+sonic(config)#
+```
+
+##### Configure sFlow agent interface
+Syntax:
+
+`sflow agent <ifname>`
+
+```
+sonic(config)# sflow agent-id Ethernet0
+sonic(config)#
+```
+
+##### Reset sFlow agent to default interface
+```
+sonic(config)# no sflow agent-id
+sonic(config)#
+```
+
+##### Configure sFlow polling-interval
+Syntax:
+
+interval: [5 - 300] (0 to disable)
+
+`sflow polling-interval <interval #>`
+
+```
+sonic(config)# sflow polling-interval 44
+sonic(config)#
+```
+
+##### Reset sFlow polling-interval to default
+```
+sonic(config)# no sflow polling-interval
+sonic(config)#
+```
+
+sFlow configurations for specific interface are executed in interview-configuration-view:
+```
+sonic# configure terminal
+sonic(config)# interface Ethernet 0
+sonic(conf-if-Ethernet0)#
+```
+
+##### Enable sFlow
+```
+sonic(conf-if-Ethernet0)# sflow enable
+sonic(conf-if-Ethernet0)#
+```
+
+##### Disable sFlow
+```
+sonic(conf-if-Ethernet0)# no sflow enable
+sonic(conf-if-Ethernet0)#
+```
+
+##### Set sampling-rate
+Syntax:
+
+rate: [256 - 8388608]
```

Comment:
I think you should specify the valid ranges for a parameter *after* the syntax specification.